screen/categories/src/main/java/com/ivy/categories/CategoriesViewModel.kt

```diff
@@ -55,6 +55,8 @@ class CategoriesViewModel @Inject constructor(
     private val baseCurrency = mutableStateOf("")
     private val categories =
         mutableStateOf<ImmutableList<CategoryData>>(persistentListOf<CategoryData>())
+    private val allCategories =
```

Comment:
We don't need this state. Just in the `@Composable getCategories`()` do filter the categories state by the query. You alsoo need to `remember(categories, query) { categories.filter { ... } }`